### PR TITLE
added MGU class

### LINKED ISSUE #3120: Feature request: Add minimal gated unit (MGU)
Feature request: Add [minimal gated unit (MGU)](https://en.wikipedia.org/wiki/Gated_recurrent_unit#Minimal_gated_unit) to [flax.linen](https://flax.readthedocs.io/en/latest/api_reference/flax.linen.html#rnn-primitives). I can open a PR if desired.

### PR DESCRIPTION
**Pull request by chiamp**
Added MGU class and resolves #3120. Check out the API docs [here](https://flax--3418.org.readthedocs.build/en/3418/api_reference/flax.linen/layers.html#flax.linen.MGUCell).

### CODE REVIEW

File: flax/linen/recurrent.py [hunk 1]
```diff
@@ -500,6 +500,107 @@ def num_feature_axes(self) -> int:
...
+      the hidden state (default: initializers.orthogonal()).
+    bias_init: initializer for the bias parameters (default: initializers.zeros_init())
+    dtype: the dtype of the computation (default: None).
+    param_dtype: the dtype passed to parameter initializers (default: float32).
+  """
+  features: int
+  gate_fn: Callable[..., Any] = sigmoid
+  activation_fn: Callable[..., Any] = tanh
+  kernel_init: initializers.Initializer = default_kernel_init
+  recurrent_kernel_init: initializers.Initializer = initializers.orthogonal()
```
Comment (cgarciae) [inline flax/linen/recurrent.py:535]: interesting, can you point to the source for this?

File: tests/linen/linen_test.py [hunk 2]
```diff
@@ -712,37 +712,52 @@ def test_lstm(self):
         },
     )
 
-  def test_gru(self):
-    gru = nn.GRUCell(features=4)
+  @parameterized.parameters(
```
Comment (cgarciae) [inline tests/linen/linen_test.py:715]: Nice!

File: flax/linen/recurrent.py (hunk 1 above)
Comment (chiamp) [reply flax/linen/recurrent.py:535]: I copied it over from the GRU class' [`recurrent_kernel_init`](https://github.com/google/flax/pull/3418/files#diff-32512a9b0d5285a594cec41f03de1a4530fd5920847ca3c095fb9d0f0677ae23R423).

File: flax/linen/recurrent.py [hunk 3]
```diff
@@ -500,6 +500,112 @@ def num_feature_axes(self) -> int:
...
+    f = self.gate_fn(dense_i(name='if', bias_init=self.forget_bias_init)(inputs) + dense_h(name='hf')(h))
+    # add bias because the linear transformations aren't directly summed.
+    n = self.activation_fn(
+        dense_i(name='in', bias_init=self.activation_bias_init)(inputs) + f * dense_h(name='hn', use_bias=True)(h)
+    )
+    new_h = (1.0 - f) * n + f * h
+    return new_h, new_h
+
+  @nowrap
+  @deprecation_method_decorator
```
Comment (cgarciae) [inline flax/linen/recurrent.py:589]: Lets merge #3425 before this PR so we can remove this decorator.

**Review by cgarciae — APPROVED** [review]

File: flax/linen/recurrent.py (hunk 3 above)
Comment (chiamp) [reply flax/linen/recurrent.py]: done

**Pull request merged**